File: lab1.m
clc;
clear all;
close all;
bw=30000; %bw in kHz
schannel_bw=25; %single_channel bw in kHz
disp('Channel Bandwidth..(unit in kHz/duplex channel)');
dup_ch_bw=2*schannel_bw;
t_ch=(bw/dup_ch_bw);
disp(dup_ch_bw);
disp('Total available channel');
disp(t_ch);
cc_bw=1000; %control channel bw in kHz
t_cc=cc_bw/dup_ch_bw;
disp('Total control channel');
disp(t_cc);
N=[4 7 12];
for i=1:3
    ch=t_ch/N(i);
    ch_per_cell=round(ch);
    %disp('Channel per cell')
    fprintf('For n = %d, total number of channel available per cell =', N(i));
    %disp(N(i));
    disp(ch_per_cell);
    c=(t_cc/N(i));
    cc=round(c);
    v=(t_ch-t_cc)/N(i);
    vc=round(v);
    disp('Control channel and voice channel are..');
    disp(cc);
    disp(vc);
end

File: lab_1.py
print('------------------------------------------------')
bw = 30e3
single_channel_bw = 25

# duplex channel bandwidth
duplex_channel_bw = 2 * single_channel_bw
print('Channel bw =', duplex_channel_bw)

# total channel
total_channel = bw // duplex_channel_bw
print('Total available channel =', total_channel)

# control channel bandwidth
controlChannel_bw = 1000

# total control channel
total_controlChannel = controlChannel_bw // duplex_channel_bw
print('Total Control Channel =', total_controlChannel)

# Array of different value for N(number of cells)
print('------------------------------------------------')
print('For various cluster size')
print('------------------------------------------------')
N = [4, 7, 12]
for cluster_size in N:
    # cluster size
    print('Cluster Size=', cluster_size)

    # channel per cells
    channel_per_cell = round(total_channel / cluster_size)
    print('Channel per cell =', channel_per_cell)

    # control channels per cell
    controlChannel_per_cell = round(total_controlChannel / cluster_size)
    print('Control channel =', controlChannel_per_cell)

    # voice channel per cell
    voiceChannel = round((total_channel - total_controlChannel) / cluster_size)
    print('Voice channel =', voiceChannel)
    print('------------------------------------------------')